ksqldb-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonSerdeFactory.java

```diff
@@ -40,13 +43,19 @@
 class KsqlJsonSerdeFactory {
 
   private final boolean useSchemaRegistryFormat;
+  private final JsonSchemaProperties properties;
+
+  KsqlJsonSerdeFactory() {
+    useSchemaRegistryFormat = false;
+    properties = null;
+  }
 
   /**
-   * @param useSchemaRegistryFormat whether or not to require the magic byte and
-   *                                schemaID as part of the JSON message
+   * @param properties JsonSchemaFormat properties
    */
-  KsqlJsonSerdeFactory(final boolean useSchemaRegistryFormat) {
-    this.useSchemaRegistryFormat = useSchemaRegistryFormat;
+  KsqlJsonSerdeFactory(final JsonSchemaProperties properties) {
+    this.useSchemaRegistryFormat = true;
+    this.properties = Objects.requireNonNull(properties, "properties");
   }
```

Comment:
I have a suggestion for better readability of this class. What do you think about keeping the old constructor as private and create these two static methods:
```
// returns a KsqlJsonSerdeFactory with SR format enabled
final KsqlJsonSerdeFactory factory = KsqlJsonSerdeFactory.withSchemaRegistryProperties(properties);

// returns a instance of KsqlJsonSerdeFactory without SR disabled
final KsqlJsonSerdeFactory factory = KsqlJsonSerdeFactory.get();
```

Reply:
Sounds good to me!